File: dugout-api/src/main/kotlin/com/dugout/api/domain/attendance/entity/AttendanceStatus.kt
package com.dugout.api.domain.attendance.entity

/**
 * 출석 투표 상태 5종.
 * - ATTEND: 참가 ✅
 * - ABSENT: 불참 ❌
 * - MAYBE: 미정 ❓
 * - LATE: 늦참 ⏰ (경기 시작 후 합류)
 * - EARLY_LEAVE: 조퇴 🚪 (경기 시작은 참석하나 중간 이탈)
 */
enum class AttendanceStatus {
    ATTEND,
    ABSENT,
    MAYBE,
    LATE,
    EARLY_LEAVE,
}

val AttendanceStatus.isAvailable: Boolean
    get() = this == AttendanceStatus.ATTEND ||
        this == AttendanceStatus.LATE ||
        this == AttendanceStatus.EARLY_LEAVE

fun isMeaningfulAttendanceChange(
    previous: AttendanceStatus,
    new: AttendanceStatus,
): Boolean = previous.isAvailable != new.isAvailable

File: dugout-ios/Features/Match/Sources/Domain/Entities/AttendanceStatus.swift
//
//  AttendanceStatus.swift
//  DugoutMatchFeature
//
//  백엔드 AttendanceStatus enum (ATTEND/ABSENT/MAYBE/LATE/EARLY_LEAVE) 대응.
//

import Foundation

public enum AttendanceStatus: String, Sendable, Codable, CaseIterable, Hashable {
    case attend     = "ATTEND"
    case absent     = "ABSENT"
    case maybe      = "MAYBE"
    case late       = "LATE"
    case earlyLeave = "EARLY_LEAVE"

    public var koreanLabel: String {
        switch self {
        case .attend: "참가"
        case .absent: "불참"
        case .maybe: "미정"
        case .late: "늦참"
        case .earlyLeave: "조퇴"
        }
    }

    public var emoji: String {
        switch self {
        case .attend: "✅"
        case .absent: "❌"
        case .maybe: "❓"
        case .late: "⏰"
        case .earlyLeave: "🚪"
        }
    }
}
